circuits/src/zk_circuits/proof_linking.rs

```diff
@@ -259,3 +259,476 @@ where
+
+        let (comm_witness, comm_statement) =
+            commitments_witness_statement(&wallet, &public_share, &private_share);
+
+        test_commitments_reblind_link(
+            reblind_witness,
+            reblind_statement,
+            comm_witness,
+            comm_statement,
+        )
+        .unwrap();
+    }
+
+    /// Tests a valid link between a proof of VALID COMMITMENTS and a proof of
+    /// VALID MATCH SETTLE on the first party's side
+    #[tokio::test]
+    async fn test_commitments_match_settle_valid_party0() {
+        let (comm_witness, comm_statement, match_settle_witness, match_settle_statement) =
+            build_commitments_match_settle_data(PARTY0);
+
+        test_commitments_match_settle_link(
+            PARTY0,
+            comm_witness,
+            comm_statement,
+            match_settle_witness,
+            match_settle_statement,
+        )
+        .await
+        .unwrap()
+    }
+
+    /// Tests a valid link between a proof of VALID COMMITMENTS and a proof of
+    /// VALID MATCH SETTLE on the second party's side
+    #[tokio::test]
+    async fn test_commitments_match_settle_valid_party1() {
+        let (comm_witness, comm_statement, match_settle_witness, match_settle_statement) =
+            build_commitments_match_settle_data(PARTY1);
+
+        test_commitments_match_settle_link(
+            PARTY1,
+            comm_witness,
+            comm_statement,
+            match_settle_witness,
+            match_settle_statement,
+        )
+        .await
+        .unwrap();
+    }
+
+    /// Tests an invalid link between a proof of VALID COMMITMENTS and a proof
+    /// of VALID MATCH SETTLE wherein the shares are modified between the
+    /// two proofs
+    #[tokio::test]
+    #[should_panic(expected = "ProofLinkVerification")]
+    #[allow(non_snake_case)]
+    async fn test_commitments_match_settle_invalid_link__modified_shares() {
+        let (comm_witness, comm_statement, mut match_settle_witness, mut match_settle_statement) =
+            build_commitments_match_settle_data(PARTY0);
+
+        // Modify the shares on the first party's side, but keep them consistent such
+        // that they remain a valid witness
+        let mut rng = thread_rng();
+        let mut modified_shares = match_settle_witness.party0_public_shares.to_scalars();
+        let mut result_shares = match_settle_statement.party0_modified_shares.to_scalars();
+        let modification_idx = (0..modified_shares.len()).sample_single(&mut rng);
+
+        let modification = Scalar::random(&mut rng);
+        modified_shares[modification_idx] += modification;
+        result_shares[modification_idx] += modification;
+        match_settle_witness.party0_public_shares =
+            WalletShare::from_scalars(&mut modified_shares.into_iter());
+        match_settle_statement.party0_modified_shares =
+            WalletShare::from_scalars(&mut result_shares.into_iter());
```

Comment:
nit: wanna break this out into a helper, seeing as though we use it across multiple tests?

Reply:
It's only done twice, and is used differently in both cases (add to both in one, add/subtract in the other), so I don't think a helper is going to do much here and would need an odd behavioral switch. I'll leave it for now, but will add in the future if this becomes more repetitive